cxx-tests/src/tests.rs

```diff
@@ -86,30 +109,214 @@ mod bindgenish {
       }
     }
   }
+
+  unsafe impl MoveNew for Foo {
+    unsafe fn move_new(
+      mut src: Pin<moveit::MoveRef<Self>>,
+      this: Pin<&mut MaybeUninit<Self>>,
+    ) {
+      let src: &mut _ = ::std::pin::Pin::into_inner_unchecked(src.as_mut());
+      let this = std::mem::transmute(this);
```

Comment:
Done. In the process, I realized I had some UB here:
```rust
    fn bar_constructor(_this: Pin<&mut Bar>);
```
because (AIUI) we're not permitted to have a _reference_ to something that may not yet be initialized. I have changed this to:

```rust
    fn bar_constructor(_this: *mut Bar);
```

Happily this also enables zapping a couple of other transmutes.